python/lance/_lib.pyx

```diff
@@ -158,6 +161,17 @@ cdef extern from "lance/arrow/dataset.h" namespace "lance::arrow" nogil:
                 optional[uint64_t] version,
         )
 
+        CDatasetVersion version() const;
```

Comment:
How come this is not wrapped in CResult?

Reply:
Right, so this is fetching the "current version", which is converting from already loaded Manifest. There is no I/O involved , so no error case in this code path.